gusto/state.py

```diff
@@ -253,9 +254,12 @@ def dump(self, t=0, pickup=False):
                 self.dumpfile_ll.write(*self.to_dump_latlon)
 
             # compute diagnostics
+            diagnostic_fns = ['min', 'max', 'rms', 'l2']
             for name in self.diagnostics.fields:
-                data = self.diagnostics.l2(self.field_dict[name])
-                self.diagnostic_data[name]["l2"].append(data)
+                for fn in diagnostic_fns:
+                    d = getattr(self.diagnostics, fn)
+                    data = d(self.field_dict[name])
+                    self.diagnostic_data[name][fn].append(data)
```

Comment:
I think we could make diagnostic_fields a dictionary and specify diagnostic_fns for each diagnostics, e.g., diagnostic_fields = {CourantNumber():['max'], VelocityY():['max', 'rms']}. What do you think @jshipton?